### PR TITLE
[release-3.4]: .travis.yml: Test with go v1.15.11

### PR DESCRIPTION
**Pull request by lilic**
Currently, in CI the tests are only run with go v1.12, this adds also go v1.15.7.

### CODE REVIEW

**Comment by lilic** [comment]
I was not sure if there are any other places that would need updates, from what I can tell this is the only thing for CI testing as the Makefile is not used otherwise, please correct me if I am wrong, still new to the codebase. 😄🤞

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ services: docker
 
 go:
 - 1.12.12
+- 1.15.7
```
Comment (ptabor) [inline .travis.yml:10]: I would consider excluding a lot of variants
to save limited CI resources: for analogy please see: 

https://github.com/etcd-io/etcd/blob/7f97dfd45ac82518716a59c246e2c2a95c9825c7/.travis.yml#L37

I think it's sufficient to run:
 - linux-amd64-fmt
 - linux-amd64-unit
 - linux-amd64-integration-4-cpu
 - linux-amd64-functional
 - linux-amd64-grpcproxy
 - all-build

Comment (lilic) [reply .travis.yml:10]: Got it, makes sense will do!

This reminds me actually, about the actual version, I wanted to use the latest go 1.15.x which is v1.15.11 but it seems there is no tagged image version for this go version, how does that work?

**Review by ptabor — APPROVED** [review]

**Comment by lilic** [comment]
Thank you @ptabor for your review and approval! So tests were green for go 1.15 and go 1.12 in the:
* https://travis-ci.com/github/etcd-io/etcd/builds/223180853 

Sadly now they failed twice in a row, I did notice a few flakes, around `TestBalancerUnderNetworkPartitionWatchFollower `, which I cannot reproduce locally ever. I don't have the ability to just retest the individual failures so I had to force push. 🤞 🟢 

For the TLS cypher suites tests sadly the direct cherry-pick of `60e44286fa3c0c0` did not work due to go 1.12 not having the `tls.CipherSuites()` function. I solved this by requiring two files, which one is used depends on the build tags go versions, I am open to suggestions on a different approach but I rather played it safe here. The go 1.15 files are aligned to what we have in master and others were kept the same way.

I still have the open question around the latest go version:
I wanted to use the latest go 1.15.x which is v1.15.11 but it seems there is no tagged image version for this go version, how does that work? I can also bump in a follow up PR and we leave it as is for now.

**Comment by ptabor** [comment]
> TestBalancerUnderNetworkPartitionWatchFollower

FYI: I hope I fixed this flake in master in: a57e967d84fa80eca8aacd785cf6cd915c75cfdd

**Comment by lilic** [comment]
Thanks for the heads up @ptabor! Have no hit that one anymore it was mainly `TestV3WatchCancellation` for the past few runs so I cherry-picked one of your commits from master. But can cherry-pick the commit as well next time I hit `TestBalancerUnderNetworkPartitionWatchFollower `! 

Also, it's green, please have another look, thank you! 🎉

**Pull request merged**